x-pack/plugins/security_solution/scripts/run_cypress/utils.ts

```diff
@@ -52,28 +48,56 @@ export const parseTestFileConfig = (
       return {};
     }
 
-    return _.reduce(
-      ftrConfig.value.properties,
-      (acc: Record<string, string | number | Record<string, string>>, property) => {
-        const key = (property.key as Identifier).name;
-        let value;
-        if (property.value.type === 'ArrayExpression') {
-          value = _.map(property.value.elements, (element) => {
-            if (element.type === 'StringLiteral') {
-              return element.value as string;
-            }
-            return element.value as string;
-          });
-        } else if (property.value.type === 'StringLiteral') {
-          value = property.value.value;
-        }
-        if (key && value) {
-          acc[key] = value;
-        }
-        return acc;
-      },
-      {}
-    );
+    const ftrConfigCode = generate(ftrConfig.value, { jsonCompatibleStrings: true }).code;
+
+    try {
+      // TODO:PT need to assess implication of using this approach to get the JSON back out
+      const ftrConfigJson = new Function(`return ${ftrConfigCode}`)();
+      return TestFileFtrConfigSchema.validate(ftrConfigJson);
+    } catch (err) {
+      throw new Error(
+        `Failed to parse 'ftrConfig' value defined in 'describe()' at ${filePath}. ${err.message}\nCode: ${ftrConfigCode}`
+      );
+    }
   }
-  return undefined;
+
+  return {};
 };
+
+const TestFileFtrConfigSchema = schema.object(
+  {
+    license: schema.maybe(schema.string()),
+    enableExperimental: schema.maybe(
+      schema.arrayOf(
+        schema.string({
+          validate: (value) => {
+            const allowedValues = getExperimentalAllowedValues();
+
+            if (!allowedValues.includes(value)) {
+              return `Invalid [enableExperimental] value {${value}.\nValid values are: [${allowedValues.join(
+                ', '
+              )}]`;
+            }
+          },
+        })
+      )
+    ),
+    productTypes: schema.maybe(
+      // TODO:PT write validate function to ensure that only the correct combinations are used
+      schema.arrayOf(
+        schema.object({
+          product_line: schema.oneOf([
+            schema.literal('security'),
+            schema.literal('endpoint'),
+            schema.literal('cloud'),
+          ]),
+
+          product_tier: schema.oneOf([schema.literal('essentials'), schema.literal('complete')]),
+        })
+      )
```

Comment:
I don't feel very comfortable having this serverless-specific config definition duplicated in the security_solution plugin, this schema is already defined in the security_solution_serverless plugin. 
Not sure if there's any other approach to have this test config schema defined, but since this is testing utils, probably it is just fine.

Reply:
I might be able to import and use the schema from the serverless plugin. Let me try

Reply:
I imported it from security solution serverless plugin and will see if the build fails (worried about circular dependencies). If it does fail, then one other alternative would be to move the `productTypes` (only) schema to `security_solution/common/**`. How do you feel about that?

Reply:
@semd - importing it from security solution serverless errors due to circular dependency. I think he only other option is to a) not validate it or b) move the `productTypes` schema to security solution common. Let me know what you think and I'll include it in my next PR